applications/GeoMechanicsApplication/custom_utilities/process_factory.cpp

```diff
@@ -0,0 +1,38 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "custom_utilities/process_factory.hpp"
+#include "includes/kratos_parameters.h"
+
+
+namespace Kratos
+{
+
+ProcessFactory::ProductType ProcessFactory::Create(const std::string& rProcessClassName,
+                                                   const Parameters&  rProcessSettings) const
+{
+    auto pos = mCreatorMap.find(rProcessClassName);
+    if (pos == mCreatorMap.end()) {
+        return nullptr;
+    }
+
+    return pos->second ? pos->second(rProcessSettings) : nullptr;
+}
+
+void ProcessFactory::AddCreator(const std::string&                            rProcessClassName,
+                                std::function<ProductType(const Parameters&)> Creator)
+{
+    mCreatorMap[rProcessClassName] = std::move(Creator);
```

Comment:
I'm not sure if it's feasible, but for the key, we might be able to use the Process::Info() function, such that the key is always in accordance with the process (unless we need the flexibility to have custom names/keys)

Reply:
I have opted to give the client of the factory some flexibility with respect to the adopted names and to not enforce the result returned by `Process::Info()`.